pkg/ddevapp/ddevapp.go

```diff
@@ -507,12 +507,16 @@ func (app *DdevApp) ImportDB(imPath string, extPath string, progress bool, noDro
 	}
 
 	// The perl manipulation removes statements like CREATE DATABASE and USE, which
-	// throw off imports.
-	inContainerCommand := fmt.Sprintf(`mysql -uroot -proot -e "%s" && pv %s/*.*sql | perl -p -e 's/^(CREATE DATABASE|USE|.*DROP DATABASE).*$//' | mysql %s`, preImportSQL, insideContainerImportPath, targetDB)
+	// throw off imports. This is a scary manipulation, as it must not match actual content
+	// as has actually happened with https://www.ddev.com/ddev-local/ddev-local-database-management/
+	// and in https://github.com/drud/ddev/issues/2787
+	// The backtick after USE is inserted via fmt.Sprintf argument because it seems there's
+	// no way to escape a backtick in a string literal.
+	inContainerCommand := fmt.Sprintf(`mysql -uroot -proot -e "%s" && pv %s/*.*sql | perl -p -e 's/^(CREATE DATABASE \/\*|USE %s)[^;]*;//' | mysql %s`, preImportSQL, insideContainerImportPath, "`", targetDB)
 
 	// Handle the case where we are reading from stdin
 	if imPath == "" && extPath == "" {
-		inContainerCommand = fmt.Sprintf(`mysql -uroot -proot -e "%s" && perl -p -e 's/^(CREATE DATABASE|USE|.*DROP DATABASE).*$//' | mysql %s`, preImportSQL, targetDB)
+		inContainerCommand = fmt.Sprintf(`mysql -uroot -proot -e "%s" && perl -p -e 's/^(CREATE DATABASE \/\*|USE %s)[^;]*;//' | mysql %s`, preImportSQL, "`", targetDB)
```

Comment:
do we still desire to remove `DROP DATABASE`? I'm wondering if those can be dangerous too. I can imagine a scenario when a user has a number of sites, intends to import to a site `A` but the SQL dump contains `DROP DATABASE b;` and what appeared to be an unrelated site for the `ImportDB` suddenly stops working.

Reply:
I looked at several dumps and did not find any use of `DROP DATABASE` in a standard dump. However... in a case like this it would be DROP DATABASE IF EXISTS... and the database would not exist. 

But your point about breaking a *different* site is interesting. 

Still, one would expect that if DROP DATABASE *were* in there, it would refer to the current database, so the current site, so would not do harm?

Reply:
I acknowledge it's a hypothetical maybe type of scenario that we probably don't need to worry about here. Still wondering if for ddev-live we may need to parse AST, modify there and then generate back the SQL dump to eliminate even any hypothetical scenarios.

Reply:
IMO this is all bad. We shouldn't be doing sed/perl/awk against a db dump. I imagine that we could figure out a way to do this only to the header section of the dump.

I hope you noticed some of the limiting factors now added to this version - using [^;]* instead of .* for the removal, and making sure it was anchored.